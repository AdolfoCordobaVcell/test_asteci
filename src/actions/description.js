import { Fetch } from '../helper/fetch';
import { types } from '../types/types';

export const actionOpen = ( id ) => {
    let data;

    return async (dispatch) => {
        try {
            const resp = await Fetch(`?_id=${ id }`);

            const body =  await resp.json();

            data = body.results;

            dispatch( open(data) )
        } catch (error) {
            console.log(error);
        }
    }
}

export const open = (event) => ({
    type: types.open,
    payload: event
});