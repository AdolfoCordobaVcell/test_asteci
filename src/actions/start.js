import { Fetch } from '../helper/fetch';
import { types } from '../types/types';

export const actionStart = ( one, two, setClima, setTotal ) => {
    let data;

    return async (dispatch) => {
        try {
            const resp = await Fetch('');

            const body =  await resp.json();

            data = body.results.slice(one, two);

            setTotal(body.results.length);

            setClima(data);

            dispatch( start(data) );
        } catch (error) {
            console.log(error);
        }
    }
}

const start = (event) => ({
    type: types.start,
    payload: event
});