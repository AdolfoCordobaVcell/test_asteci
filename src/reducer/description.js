import {types} from '../types/types';

const initialState = {
    event: [],
}

export const description = (state = initialState, action) => {
    switch (action.type) {
        case types.open:
            return {
                ...state,
                event: action.payload
            }

        default: 
            return state;
    }
}