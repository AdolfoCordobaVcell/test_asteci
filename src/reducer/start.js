import {types} from '../types/types';

const initialState = {
    event: [],
}

export const start = (state = initialState, action) => {
    switch (action.type) {
        case types.start:
            return {
                ...state,
                event: action.payload
            }

        default: 
            return state;
    }
}