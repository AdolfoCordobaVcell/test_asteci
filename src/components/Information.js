import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { actionStart } from '../actions/start';
import { actionOpen } from '../actions/description';

const Information = () => {
    const [total, setTotal] = useState(0);
    const [clima, setClima] = useState([]);
    const [one, setOne] = useState(0);
    const [two, setTwo] = useState(10);

    const dispatch = useDispatch();

    const navigate = useNavigate();

    useEffect(() => {
        dispatch( actionStart( one, two, setClima, setTotal ) );
    }, [one, two, setClima, setTotal, dispatch]);

    const description = ( id ) => {
        navigate("/description");

        dispatch(actionOpen(id));
    }

    const anterior = () => {
        let counOne = one;
        let counTwo = two;
        
        counOne -= 10;
        counTwo -= 10;

        setOne(counOne);
        setTwo(counTwo);
    }

    const siguiente = () => {
        let counOne = one;
        let counTwo = two;
        
        counOne += 10;
        counTwo += 10;

        setOne(counOne);
        setTwo(counTwo);
    }

    return (
        <div>
            <table className="table">
                <thead className='table__thead'>
                    <tr>
                        <th className='table__th'>_id</th>
                        <th className='table__th'>city id</th>
                        <th className='table__th'>name</th>
                        <th className='table__th'>state</th>
                        <th className='table__th'>probability of precipt</th>
                        <th className='table__th'>relative humidity</th>
                        <th className='table__th'>lastre porttime</th>
                        <th className='table__th'>probability of precip</th>
                    </tr>
                </thead>

                <tbody className="table__tbody">
                    
                        {
                            clima.map(({_id, cityid, name, state, probabilityofprecip, relativehumidity, lastreporttime}) => (
                                <tr key={_id} className="table__tr" onClick={ () => description(_id) }>
                                    <td className="table__td">
                                    {_id}
                                    </td>

                                    <td className="table__td">
                                    {cityid}
                                    </td>

                                    <td className="table__td">
                                    {name}
                                    </td>

                                    <td className="table__td">
                                    {state}
                                    </td>

                                    <td className="table__td">
                                    {probabilityofprecip}
                                    </td>

                                    <td className="table__td">
                                    {relativehumidity}
                                    </td>

                                    <td className="table__td">
                                    {lastreporttime}
                                    </td>

                                    <td className="table__td">
                                    {(probabilityofprecip > 60 || relativehumidity > 50)
                                        ? 'yes' : 'No'}
                                    </td>
                                </tr>
                            ))
                        }
                </tbody>
            </table>

            <div className='buttons'>
                <button className='buttons__btn' onClick={ anterior } disabled={ one===0 && true}>Anterior</button>
                <button className='buttons__btn buttons__btn--sig' onClick={ siguiente } disabled={ two===100 && true}>Siguiente</button>
            </div>

            <div className='total'>
                <p>Total de registros {total} </p>
                <p>del {one + 1} al {two}</p>
            </div>
        </div>
    );
};

export default Information;