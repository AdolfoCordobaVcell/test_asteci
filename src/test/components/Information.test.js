import React from 'react';
import { mount } from "enzyme";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { actionStart } from '../../actions/start';
import { actionOpen } from '../../actions/description';
import Information from '../../components/Information';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState= {
    
}

const store = mockStore( initState );

store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <Information />
        </MemoryRouter>
    </Provider>
);

jest.mock('../../actions/start', () => ({
    actionStart: jest.fn()
}));

jest.mock('../../actions/description', () => ({
    actionOpen: jest.fn()
}));

describe('Pruebas en <Information />', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de disparar el actionStart', () => {
        wrapper.find('.buttons__btn--sig').simulate('click');

        expect( actionStart ).toHaveBeenLastCalledWith( 10, 20, expect.any(Function), expect.any(Function) );
    });

//     test('Debe de disparar el actionOpen', () => {
//         wrapper.find('.table__tr').at(1).props().onClick();

//         expect( actionOpen ).toHaveBeenLastCalledWith( expect.any(String) );
//     });
});