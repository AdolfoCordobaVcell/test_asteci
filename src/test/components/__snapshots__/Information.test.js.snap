// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Information /> Debe de mostrarse correctamente 1`] = `
<div>
  <table
    className="table"
  >
    <thead
      className="table__thead"
    >
      <tr>
        <th
          className="table__th"
        >
          _id
        </th>
        <th
          className="table__th"
        >
          city id
        </th>
        <th
          className="table__th"
        >
          name
        </th>
        <th
          className="table__th"
        >
          state
        </th>
        <th
          className="table__th"
        >
          probability of precipt
        </th>
        <th
          className="table__th"
        >
          relative humidity
        </th>
        <th
          className="table__th"
        >
          lastre porttime
        </th>
        <th
          className="table__th"
        >
          probability of precip
        </th>
      </tr>
    </thead>
    <tbody
      className="table__tbody"
    />
  </table>
  <div
    className="buttons"
  >
    <button
      className="buttons__btn"
      disabled={true}
      onClick={[Function]}
    >
      Anterior
    </button>
    <button
      className="buttons__btn buttons__btn--sig"
      disabled={false}
      onClick={[Function]}
    >
      Siguiente
    </button>
  </div>
  <div
    className="total"
  >
    <p>
      Total de registros 
      0
       
    </p>
    <p>
      del 
      1
       al 
      10
    </p>
  </div>
</div>
`;
